src/Agent.Listener/Configuration/FeatureFlagProvider.cs

```diff
@@ -9,6 +9,7 @@
 using Microsoft.VisualStudio.Services.FeatureAvailability;
 using Microsoft.VisualStudio.Services.FeatureAvailability.WebApi;
 using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
```

Comment:
(not related to this line): Need to combine with #4489

Reply:
On surface, it looks as though this change could replace mine entirely. And provides a model for me to provide a couple of additional performance-related changes. I would likely continue to evolve the agent-side ISecretMasker interface to do this. So maybe we should focus on getting this in and I can look at how/whether to modify a bit further for that work? Maybe we could protect *all* of this work under a single feature flag until it's finished (which will only be a week or two) so that we don't have to over-invest in lots of new configurability.

Reply:
@michaelcfanning , I renamed FF to `UseMaskingPerformanceEnhancements`, we can use it for all future changes.